### PR TITLE
Added ev3dev/helper.py and demo/lego_official_projects/

### PR DESCRIPTION
**Pull request by dwalton76**

### CODE REVIEW

**Comment by dwalton76** [comment]
Hey @rhempel anything else to be done with this one?

**Comment by rhempel** [comment]
Will review and commit tomorrow - thks!

**Comment by dwalton76** [comment]
Is this one ok? It has been a few weeks :)

**Comment by rhempel** [comment]
Sorry, I got distracted with another thing I was merging - will do it today...

File: ev3dev/brickpi.py [hunk 1]
```diff
@@ -33,11 +33,13 @@
 OUTPUT_B = 'ttyAMA0:outB'
 OUTPUT_C = 'ttyAMA0:outC'
 OUTPUT_D = 'ttyAMA0:outD'
+OUTPUTS = (OUTPUT_A, OUTPUT_B, OUTPUT_C, OUTPUT_D)
```
Comment (rhempel) [inline ev3dev/brickpi.py]: Remind me again why this change is necessary here and in ev3.py?

File: ev3dev/core.py [hunk 2]
```diff
@@ -182,8 +183,16 @@ def get_index(file):
                 self._device_index = get_index(name)
                 self.connected = True
             except StopIteration:
+                self._path = None
+                self._device_index = None
                 self.connected = False
 
+    def __str__(self):
```
Comment (rhempel) [inline ev3dev/core.py]: Does anyone else have comments on this? If we make this change it should be to cover a defect.

File: ev3dev/brickpi.py (hunk 1 above)
Comment (dwalton76) [inline ev3dev/brickpi.py]: It is so later you can do a sanity check to verify that an OUTPUT_X is a valid output

Comment (rhempel) [inline ev3dev/brickpi.py]: I'd rather take it out of core if it's designed to be used as a "helper". You'd get an error if you tried to open an incorrect port anyways,

**Comment by dwalton76** [comment]
@rhempel I've move INPUTS and OUTPUTS into helper.py

**Comment by dwalton76** [comment]
ping :)

File: ev3dev/core.py (hunk 2 above)
Comment (rhempel) [inline ev3dev/core.py]: What is this code doing here?

Comment (dwalton76) [inline ev3dev/core.py]: Do you mean why is it here instead of in helper.py or what does __str__() do?  Sorry, I'm not following what the question/concern is.

Comment (rhempel) [inline ev3dev/core.py]: What specific problem is this code change addressing -

Comment (dwalton76) [inline ev3dev/core.py]: Say you do the following:
`motor = LargeMotor(OUTPUT_B)`
`log.info("%s: move to position %d" % (motor, args.degrees))`


Without the \__str__ code here you will see output such as:
`2016-02-16 02:06:12,195  INFO: <ev3dev.helper.LargeMotor object at 0xb63aa270>: move to position 45`

but with this \__str__ code you will see:
`2016-02-16 02:05:07,880  INFO: LargeMotor(outB): move to position 45`

It just makes it much easier to print/log an object because you can get something that is more human readable.

Comment (dwalton76) [inline ev3dev/core.py]: @rhempel are you ok with these changes?

Comment (rhempel) [inline ev3dev/core.py]: Yes, I'll merge later today - sorry for the delays. Thanks for the explanation!

**Comment by rhempel** [comment]
Thanks @dwalton76 - this helps clean up diagnostic messages.
